### PR TITLE
CORE-18548: Prepare HTTP client in Gateway

### PR DESCRIPTION
**Pull request by yift-r3**
Adds link manager service endpoint to the gateway arguments.
Read the command-line argument and create a client to send messages to the link manager.

See API pull request in https://github.com/corda/corda-api/pull/1393

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
<p><b style="color:black" >Rule</b>: 2-PEM-PRIVATE<br>
    <b style="color:black" >Severity</b>:&nbsp;CRITICAL<br><b style="color:black">File</b>: libs/utilities/src/test/kotlin/net/corda/utilities/crypto/PublicKeyFactoryTest.kt<br>
       <b style="color:black">Lines</b>: 28<br>
    <b style="color:black" >Scanner:</b> <a href="https://github.com/cider-rnd/secret-diver">Secret Diver</a></p>
    <details>
    <summary><b style="color:black">More Info</b></summary>
    Contains a private key
    </details>

File: components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/internal/LinkManagerRpcClient.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+internal class LinkManagerRpcClient(
+    publisherFactory: PublisherFactory,
+    platformInfoProvider: PlatformInfoProvider,
+    bootConfig: SmartConfig,
+) {
+    private val client by lazy {
+        publisherFactory.createHttpRpcClient()
+    }
+    private val endpoint by lazy {
+        bootConfig.getString(P2P_LINK_MANAGER_WORKER_REST_ENDPOINT)
```
Comment (williamvigorr3) [inline components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/internal/LinkManagerRpcClient.kt:23]: Should we handle `P2P_LINK_MANAGER_WORKER_REST_ENDPOINT` not being in the boot config somehow?

File: processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/internal/GatewayProcessorImpl.kt [hunk 2]
```diff
@@ -79,9 +82,11 @@ class GatewayProcessorImpl @Activate constructor(
                     subscriptionFactory,
                     publisherFactory,
                     coordinatorFactory,
-                    configMerger.getMessagingConfig(event.config),
+                    messagingConfiguration = configMerger.getMessagingConfig(event.config),
```
Comment (williamvigorr3) [inline processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/internal/GatewayProcessorImpl.kt]: Nit: Can we add this at the end or reorder.

**Review by williamvigorr3 — COMMENTED** [review]
Looks good. I left two comments.

File: components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/internal/LinkManagerRpcClient.kt (hunk 1 above)
Comment (yift-r3) [reply components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/internal/LinkManagerRpcClient.kt:23]: I've made it unlazy, so if the URL won't be there the process will crash on startup.

File: components/gateway/src/integrationTest/kotlin/net/corda/p2p/gateway/GatewayIntegrationTest.kt [hunk 3]
```diff
@@ -124,9 +124,12 @@ internal class GatewayIntegrationTest : TestBase() {
     private val sessionId = "session-1"
     private val instanceId = AtomicInteger(0)
 
-    private val messagingConfig = SmartConfigImpl.empty()
+    private val baseMessagingConfig = SmartConfigImpl.empty()
         .withValue(INSTANCE_ID, ConfigValueFactory.fromAnyRef(instanceId.getAndIncrement()))
         .withValue(MAX_ALLOWED_MSG_SIZE, ConfigValueFactory.fromAnyRef(10000000))
+    private fun messagingConfig() =
```
Comment (williamvigorr3) [inline components/gateway/src/integrationTest/kotlin/net/corda/p2p/gateway/GatewayIntegrationTest.kt:130]: Why is this needed?

**Review by williamvigorr3 — COMMENTED** [review]
I left a comment. Otherwise LGTM.

File: components/gateway/src/integrationTest/kotlin/net/corda/p2p/gateway/GatewayIntegrationTest.kt (hunk 3 above)
Comment (yift-r3) [reply components/gateway/src/integrationTest/kotlin/net/corda/p2p/gateway/GatewayIntegrationTest.kt:130]: We have the same line of `messagingConfig.withValue(INSTANCE_ID, ConfigValueFactory.fromAnyRef(instanceId.incrementAndGet()))` in so many places in the code. I think that we could reduce it.

**Review by williamvigorr3 — DISMISSED** [review]
LGTM

**Review by ben-millar — APPROVED** [review]
LGTM, Flow, Helm and CombinedWorker changes

**Review by davidcurrie — APPROVED** [review]
LGTM (for @corda/cluster-management )

**Review by williamvigorr3 — APPROVED** [review]
LGTM
